examples/avro_generic_consumer_example/avro_generic_consumer_example.go

```diff
@@ -0,0 +1,128 @@
+// Example function-based high-level Apache Kafka consumer
+package main
+
+/**
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// consumer_example implements a consumer using the non-channel Poll() API
+// to retrieve messages and events.
+
+import (
+	"fmt"
+	"github.com/confluentinc/confluent-kafka-go/kafka"
+	"github.com/confluentinc/confluent-kafka-go/schemaregistry"
+	"os"
+	"os/signal"
+	"syscall"
+)
+
+func main() {
+
+	if len(os.Args) < 5 {
+		fmt.Fprintf(os.Stderr, "Usage: %s <schema-registry> <broker> <group> <topics..>\n",
+			os.Args[0])
+		os.Exit(1)
+	}
+
+	url := os.Args[1]
+	broker := os.Args[2]
+	group := os.Args[3]
+	topics := os.Args[4:]
+	sigchan := make(chan os.Signal, 1)
+	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)
+
+	c, err := kafka.NewConsumer(&kafka.ConfigMap{
+		"bootstrap.servers": broker,
+		// Avoid connecting to IPv6 brokers:
+		// This is needed for the ErrAllBrokersDown show-case below
+		// when using localhost brokers on OSX, since the OSX resolver
+		// will return the IPv6 addresses first.
+		// You typically don't need to specify this configuration property.
+		"broker.address.family": "v4",
+		"group.id":              group,
+		"session.timeout.ms":    6000,
+		"auto.offset.reset":     "earliest"})
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Failed to create consumer: %s\n", err)
+		os.Exit(1)
+	}
+
+	fmt.Printf("Created Consumer %v\n", c)
+
+	deser := schemaregistry.GenericAvroDeserializer{}
+	err = deser.Configure(&kafka.ConfigMap{
+		"schema.registry.url": url,
+	}, false)
```

Comment:
`false` doesn't read well. does go have enums?

Reply:
This is for clarity.
In the other non-java clients we use an enum called `MessageComponent`.
I think it's particularly important in the go case, because you're not implementing a serialization framework yet, so this is user-facing.
It's also good, because it's more extensible.